packages/core/data-transfer/types/transfer-engine.d.ts

```diff
@@ -5,6 +5,19 @@ import type { DiagnosticReporter } from '../src/engine/diagnostic';
 
 export type TransferFilterPreset = 'content' | 'files' | 'config';
 
+type SchemaMap = Record<string, Schema>;
+
+// Error resolving handler middleware for the transfer engine
+export type Next<T> = (context: T) => void | Promise<void>;
+export type Middleware<T> = (context: T, next: Next) => Promise<void> | void;
+
+export type SchemaDiffHandlerContext = {
+  diffs: Record<string, Diff[]>;
+  source: ISourceProvider;
+  destination: IDestinationProvider;
```

Comment:
In the handler definition you're defining ignoredDiffs as a context property, we need to define it here too